src/modules/document-caching/AmpDocumentCachablePlugin.ts

```diff
@@ -18,8 +18,20 @@
 import { Plugin } from 'workbox-cache-expiration';
 
 export default class AmpDocumentCachablePlugin extends Plugin {
-  constructor(config: any) {
-    super(config);
+  private allowedPages_: Array<RegExp>;
+
+  constructor(config: {
```

Comment:
Let's make the config an interface defined before the default class.

Also if it's by default required your can force all fields to be optional via TS syntax. `Partial<Config>`. (https://github.com/Microsoft/TypeScript/blob/ee25cdecbca49b2b5a290ecd65224f425b1d6a9c/lib/lib.es5.d.ts#L1354)

Reply:
Also, since this extends an interface from Workbox, let's make sure we're extending the type provided by Workbox.

If Workbox isn't providing the interface type, let's fix their definitions.

Reply:
Comment 1 is fixed.

Reply:
Re Comment 2: The next version of Workbox is being written in TS, so that should solve the problem